eclipse/ts.eclipse.ide.core/src/ts/eclipse/ide/internal/core/preferences/TypeScriptCorePreferenceInitializer.java

```diff
@@ -222,4 +218,50 @@ public static void fixEmbeddedPreference(IEclipsePreferences preferences) {
 			}
 		}
 	}
+
```

Comment:
I have a doubt. There is already a TypeScriptRepositoryManager, but it currently has a "passive" role (the preference initializer populates it from the outside). Would you keep it that way and create a new different manager, or evolve it to auto-populate with TS repositories?

Personally I would go with the second option (evolving it), also making it similar to NodejsInstallManager.